### PR TITLE
THRIFT-4331 - C++: TSSLSocket fixes

### PR DESCRIPTION
**Pull request by MartinHaimberger**
fixed issue with large messages, where waitForEvent was called
mutliple times waiting for SSL_read() to get bytes and running
in the retry timeout.

fixed issue where poll was not using the right flags.

Fixes issue THRIFT-4331 on master.

### CODE REVIEW

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp [hunk 1]
```diff
@@ -759,7 +775,9 @@ unsigned int TSSLSocket::waitForEvent(bool wantRead) {
   struct THRIFT_POLLFD fds[2];
   memset(fds, 0, sizeof(fds));
   fds[0].fd = fdSocket;
-  fds[0].events = wantRead ? THRIFT_POLLIN : THRIFT_POLLOUT;
+  // use POLLIN also on write operations too, this is needed for operations
+  // which requires read and write on the socket.
```
Comment (jeking3) [inline lib/cpp/src/thrift/transport/TSSLSocket.cpp]: Interesting, what's an example of this?

Comment (MartinHaimberger) [reply lib/cpp/src/thrift/transport/TSSLSocket.cpp]: hi jeking3,

the comment may not be perfect, but SSL operations, like SSL_connect or SSL_shutdown (where this function is also used) need to perform write and read operations on the socket and we dont know where we are so we need to listen on both socket events.

I could also introduce a new function parameter for that.

**Comment by jeking3** [comment]
If you could do me a favor and rebase on the current master, I finally got all the CI build issues resolved so I'd like to see a clean build before I merge.  There are some minor formatting (indentation) issues as well.

**Comment by jeking3** [comment]
@MartinHaimberger If you could do me a favor and rebase on the current master, I finally got all the CI build issues resolved so I'd like to see a clean build before I merge. There are some minor formatting (indentation) issues as well.

**Comment by MartinHaimberger** [comment]
rebased and indentations should be fixed

**Pull request closed without merging**